src/main/java/pl/com/itsystems/cookbook/recipe/RecipeController.java

```diff
@@ -0,0 +1,86 @@
+package pl.com.itsystems.cookbook.recipe;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import pl.com.itsystems.cookbook.category.Category;
+import pl.com.itsystems.cookbook.category.CategoryService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+@RequestMapping("/recipes")
+public class RecipeController {
+    private final RecipeService recipeService;
+    private final CategoryService categoryService;
+
+    public RecipeController(RecipeService recipeService, CategoryService categoryService) {
+        this.recipeService = recipeService;
+        this.categoryService = categoryService;
+    }
+
+    @GetMapping
+    public String getAllRecipes(@RequestParam(required = false) Long category, Model model) {
+        List<Recipe> recipes = new ArrayList<>();
+        model.addAttribute("categories", categoryService.getAllCategories());
+        if (category == null) {
+            recipes = recipeService.getAllRecipes();
+            model.addAttribute("recipes", recipes);
+        } else {
+            recipes = recipeService.findByCategory(category);
+            model.addAttribute("recipes", recipes);
+        }
+        return "recipes";
+    }
+
+    @GetMapping("/{id}")
+    public String getRecipeById(@PathVariable Long id, Model model) {
+        Recipe recipe = recipeService.getRecipeById(id);
+        model.addAttribute("recipe", recipe);
+        return "recipe";
+    }
+
+    @GetMapping("/new")
+    public String createRecipeForm(Model model) {
+        model.addAttribute("recipe", new Recipe());
+        model.addAttribute("categories", categoryService.getAllCategories());
+        return "create_recipe";
+    }
+
+    @PostMapping
+    public String saveRecipe(@ModelAttribute Recipe recipe) {
+        recipeService.saveRecipe(recipe);
+        return "redirect:/recipes";
+    }
+
+    @GetMapping("/edit/{id}")
+    public String editRecipeForm(@PathVariable Long id, Model model) {
+        Recipe recipe = recipeService.getRecipeById(id);
+        model.addAttribute("recipe", recipe);
+        model.addAttribute("categories", categoryService.getAllCategories());
+        return "edit_recipe";
+    }
+
+    @PostMapping("/{id}")
+    public String updateRecipe(@PathVariable Long id, @ModelAttribute Recipe recipe) {
+        recipe.setId(id);
+        recipeService.saveRecipe(recipe);
+        return "redirect:/recipes";
+    }
+
+    @PostMapping("/{id}/like")
+    public String updateRecipeLikeIt(@PathVariable Long id) {
+        Recipe recipe = recipeService.getRecipeById(id);
+        recipe.likeIt();
+        recipeService.saveRecipe(recipe);
```

Comment:
Warstwę service wprowadzamy po to, żeby tutaj nie było już żadnej logiki. Czyli zamiast tych 3 linijek, powinieneś mieć tylko np. `recipeService.like(id)`